aws/resource_aws_cognito_user_pool_domain.go

```diff
@@ -0,0 +1,170 @@
+		Delete: resourceAwsCognitoUserPoolDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"user_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"aws_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cloudfront_distribution_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"s3_bucket": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCognitoUserPoolDomainCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cognitoidpconn
+	log.Print("[DEBUG] Creating Cognito User Pool Domain")
+
+	domain := d.Get("domain").(string)
+
+	params := &cognitoidentityprovider.CreateUserPoolDomainInput{
+		Domain:     aws.String(domain),
+		UserPoolId: aws.String(d.Get("user_pool_id").(string)),
+	}
+
+	_, err := conn.CreateUserPoolDomain(params)
+	if err != nil {
+		return fmt.Errorf("Error creating Cognito User Pool Domain: %s", err)
+	}
+
+	d.SetId(domain)
+
+	stateConf := resource.StateChangeConf{
+		Pending: []string{
+			cognitoidentityprovider.DomainStatusTypeCreating,
+			cognitoidentityprovider.DomainStatusTypeUpdating,
+		},
+		Target: []string{
+			cognitoidentityprovider.DomainStatusTypeActive,
+			cognitoidentityprovider.DomainStatusTypeFailed,
+		},
+		Timeout: 1 * time.Minute,
+		Refresh: func() (interface{}, string, error) {
+			domain, err := conn.DescribeUserPoolDomain(&cognitoidentityprovider.DescribeUserPoolDomainInput{
+				Domain: aws.String(d.Get("domain").(string)),
+			})
+			if err != nil {
+				return 42, "", err
+			}
+
+			desc := domain.DomainDescription
+
+			return domain, *desc.Status, nil
+		},
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsCognitoUserPoolDomainRead(d, meta)
+}
+
+func resourceAwsCognitoUserPoolDomainRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cognitoidpconn
+	log.Printf("[DEBUG] Reading Cognito User Pool Domain: %s", d.Id())
+
+	domain, err := conn.DescribeUserPoolDomain(&cognitoidentityprovider.DescribeUserPoolDomainInput{
+		Domain: aws.String(d.Get("domain").(string)),
+	})
+	if err != nil {
+		if isAWSErr(err, "ResourceNotFoundException", "") {
+			log.Printf("[WARN] Cognito User Pool Domain %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	desc := domain.DomainDescription
+
+	d.Set("aws_account_id", desc.AWSAccountId)
+	d.Set("cloudfront_distribution_arn", desc.CloudFrontDistribution)
+	d.Set("s3_bucket", desc.S3Bucket)
+	d.Set("user_pool_id", desc.UserPoolId)
+	d.Set("version", desc.Version)
```

Comment:
Nitpick: What do you think of checking that these attributes are actually set, in the related test?